pallets/portfolio/src/lib.rs

```diff
@@ -261,14 +301,24 @@ decl_module! {
             num: PortfolioNumber,
             to_name: PortfolioName,
         ) -> DispatchResult {
-            let sender = ensure_signed(origin)?;
-            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            let PermissionedCallOriginData {
+                primary_did,
+                secondary_key,
+                ..
+            } = Identity::<T>::ensure_origin_call_permissions(origin)?;
+            if let Some(sk) = secondary_key {
+                // Check that the secondary signer is allowed to work with this portfolio.
+                ensure!(
+                    sk.has_portfolio_permission(iter::once(num)),
```

Comment:
I wonder if we should get rid of protfolio_perms in favor of "custodianship" being introduced in #518

Reply:
I think that these are quite different concepts and we can keep both. One if for a user to permission their own secondary keys to different portfolios, and the other is to hand over control / custody to a separate identity.